python/lsst/afw/coord/refraction.py

```diff
@@ -0,0 +1,284 @@
+    return refractionStart - refractionEnd
+
+
+def deltaN(wavelength, weather):
+    """Calculate the differential refractive index of air.
+
+    The differential refractive index is the difference of the refractive index from 1.,
+    multiplied by 1E8 to simplify the notation and equations.
+    Calculated as (n_air - 1)*10^8
+
+    This replicates equation 14 of Stone 1996 "An Accurate Method for Computing Atmospheric Refraction"
+
+    Parameters
+    ----------
+    wavelength : float
+        wavelength is in nanometers
+    weather : lsst.afw.coord Weather
+        Class containing the measured temperature, pressure, and humidity
+        at the observatory during an observation
+
+    Returns
+    -------
+    float
+        The difference of the refractive index of air from 1., calculated as (n_air - 1)*10^8
+    """
+    waveNum = 1E3/wavelength  # want wave number in units 1/micron
+
+    dryAirTerm = 2371.34 + (683939.7/(130. - waveNum**2.)) + (4547.3/(38.9 - waveNum**2.))
+
+    wetAirTerm = 6487.31 + 58.058*waveNum**2. - 0.71150*waveNum**4. + 0.08851*waveNum**6.
+
+    return (dryAirTerm*densityFactorDry(weather) +
+            wetAirTerm*densityFactorWater(weather))
+
+
+def densityFactorDry(weather):
+    """Calculate dry air pressure term to refractive index calculation.
+
+    This replicates equation 15 of Stone 1996 "An Accurate Method for Computing Atmospheric Refraction"
+
+    Parameters
+    ----------
+    weather : lsst.afw.coord Weather
+        Class containing the measured temperature, pressure, and humidity
+        at the observatory during an observation
+
+    Returns
+    -------
+    float
+        Returns the relative density of dry air at the given pressure and temperature.
+    """
+    temperature = _extractTemperature(weather, useKelvin=True)
+    waterVaporPressure = humidityToPressure(weather)
+    airPressure = _extractPressure(weather)
+    dryPressure = airPressure - waterVaporPressure
+
+    eqn = (dryPressure/cds.mbar)*(57.90E-8 - 9.3250E-4*u.Kelvin/temperature +
+                                  0.25844*u.Kelvin**2/temperature**2.)
+
+    densityFactor = float((1. + eqn)*(dryPressure/cds.mbar)/(temperature/u.Kelvin))
+
+    return densityFactor
+
+
+def densityFactorWater(weather):
+    """Calculate water vapor pressure term to refractive index calculation.
+
+    This replicates equation 16 of Stone 1996 "An Accurate Method for Computing Atmospheric Refraction"
+
+    Parameters
+    ----------
+    weather : lsst.afw.coord Weather
+        Class containing the measured temperature, pressure, and humidity
+        at the observatory during an observation
+
+    Returns
+    -------
+    float
+        Returns the relative density of water vapor at the given pressure and temperature.
+    """
+    temperature = _extractTemperature(weather, useKelvin=True)
+    waterVaporPressure = humidityToPressure(weather)
+
+    densityEqn1 = float(-2.37321E-3 + 2.23366*u.Kelvin/temperature -
```

Comment:
Why the explicit cast? Here and below.

Reply:
At one point my prototype code required the cast, but it doesn't any more. I'll remove these.